tests/fixtures.py

```diff
@@ -83,8 +83,10 @@ def setUp(self):
 
 
 # Except for python/mypy#731, prefer to define
-# FilesDef = Dict[str, Union['FilesDef', str]]
-FilesDef = Dict[str, Union[Dict[str, Union[Dict[str, str], str]], str]]
+# FilesDef = Dict[str, Union['FilesDef', str, bytes]]
+FilesDef = Dict[
+    str, Union[Dict[str, Union[Dict[str, Union[str, bytes]], str, bytes]], str, bytes]
+]
```

Comment:
Let's not complicate this type with bytes. I don't see anything below that can't be handled by strings.

Reply:
I use the bytes version in `EggInfoPkgPipInstalledNoModules` to add a `top_level.txt` file that consists of a single newline:
```python
            # top_level.txt correctly reflects that no modules are installed
            "top_level.txt": b"\n",
```
Passing as a string here causes the newline to be stripped by the `DALS()` helper that is applied to string values in `build_files()`. I do test that the newline is preserved in `APITests.test_read_text()` (which is how I discovered the involvement of `DALS()`).

One thing I mentioned in #115 (but did not really touch in this PR) is the existence of a top_level.txt that is empty. This is currently treated as if the `top_level.txt` did not exist (i.e. we fall back to `_top_level_inferred()`). The `top_level.txt` with the single newline does reflect what `qiskit` ships with, and if we were to change the behavior in this area, then `EggInfoPkgPipInstalledNoModules` could serve as a relevant test case.

Reply:
Oh! Thanks for the explanation. That's rather annoying that we have to hack in bytes support to bypass the indentation helper DALS. I think I'd rather address this concern head-on in order to retain simplicity of the FilesDef. I'm thinking something like a literal wrapper, `"top_level.txt": Literal("\n"),`, that DALS would ignore, or perhaps bypassing the file builder for this one file.

Or maybe this project should adopt [the technique](https://github.com/python/importlib_resources/blob/main/importlib_resources/tests/_path.py) I've used across other projects and which has been vendored into importlib_resources, but requires explicit invocation of DALS for every file. That version has bytes support and somehow doesn't require the messy type definition (maybe because of `# type: ignore`).

Given this mess, and given that your approach is converging on what FilesSpec does, let's go with this as implemented, and I'll revisit any lingering concerns I have separately.